corehq/apps/domain/decorators.py

```diff
@@ -191,6 +193,27 @@ def wrapper(request, *args, **kwargs):
     return real_decorator
 
 
+def oauth2_check():
```

Comment:
Do I have it right that this doesn't need to check for domain access because it's only to be used inside `_login_or_challenge`?  Could also mark it private if it's not expected to be used elsewhere.

Reply:
that's right. it follows the same pattern of all those functions which have a helper just to handle auth but don't do any other checks. marking it private is a good call. ef5b39ab5f16658d046c4021d7a065f7de0a5fa9